Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40540&attempt_id=b85d304e-2086-4bbd-b29d-3626472b2206)

Attempted: i7z

<details><summary>Partial log (click to expand)</summary><p>

```
     asm volatile ("mov %%ebx, %%edi;" // save ebx (for PIC)
     ^~~
helper_functions.c:96:5: error: output number 1 not directly addressable
helper_functions.c:96:5: error: unknown register name 'edi' in 'asm'
     asm volatile ("mov %%ebx, %%edi;" // save ebx (for PIC)
     ^~~
helper_functions.c:96:5: error: output number 1 not directly addressable
make: *** [<builtin>: helper_functions.o] Error 1
builder for '/nix/store/1r4k62rqhgzpg05g737si728b8amf71j-i7z-0.27.2.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/1r4k62rqhgzpg05g737si728b8amf71j-i7z-0.27.2.drv' failed
```
</p></details>

--- a/pkgs/os-specific/linux/i7z/default.nix
+++ b/pkgs/os-specific/linux/i7z/default.nix
@@ -1,4 +1,5 @@
-{ stdenv, fetchurl, qt4, ncurses }:
+{ stdenv, lib, fetchurl, ncurses
+, withGui ? false, qt4 ? null }:
 
 stdenv.mkDerivation rec {
   name = "i7z-0.27.2";
@@ -8,29 +9,46 @@ stdenv.mkDerivation rec {
     sha256 = "1wa7ix6m75wl3k2n88sz0x8cckvlzqklja2gvzqfw5rcfdjjvxx7";
   };
 
-  buildInputs = [ qt4 ncurses ];
+  buildInputs = [ ncurses ] ++ lib.optional withGui qt4;
+
+  enableParallelBuilding = true;
 
   buildPhase = ''
+    runHook preBuild
+
     make
-    cd GUI
-    qmake
-    make clean
-    make
-    cd ..
+    ${lib.optionalString withGui ''
+      cd GUI
+      qmake
+      make clean
+      make
+      cd ..
+    ''}
+
+    runHook postBuild
   '';
 
   installPhase = ''
-    mkdir -p $out/sbin
+    runHook preInstall
+
+    mkdir -p $out/{bin,sbin}
     make install prefix=$out
-    install -Dm755 GUI/i7z_GUI $out/sbin/i7z-gui
+    ${lib.optionalString withGui ''
+      install -Dm755 GUI/i7z_GUI $out/bin/i7z-gui
+    ''}
+    mv $out/sbin/* $out/bin/
+    rmdir $out/sbin
+
+    runHook postInstall
   '';
 
-  meta = {
+  meta = with lib; {
     description = "A better i7 (and now i3, i5) reporting tool for Linux";
     homepage = https://github.com/ajaiantilal/i7z;
     repositories.git = https://github.com/ajaiantilal/i7z.git;
-    license = stdenv.lib.licenses.gpl2;
-    maintainers = [ stdenv.lib.maintainers.bluescreen303 ];
-    platforms = stdenv.lib.platforms.linux;
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ bluescreen303 ];
+    # broken on ARM
+    platforms = [ "x86_64-linux" ];
   };
 }

